---
title: "class09"
format: pdf
---

Download a CSV file from the PDB site (accessible from “Analyze” \> “PDB Statistics” \> “by Experimental Method and Molecular Type”. Move this CSV file into your RStudio project and use it to answer the following questions:

```{r}
pdb <- read.csv("pdbdata.csv")
pdb
```

## Q1: What percentage of structures in the PDB are solved by X-Ray and Electron Microscopy.

(182348 + 18817) / 215684 = 93.27%

## Q2: What proportion of structures in the PDB are protein?

(161663 + 9348 + 8404) / 215684 = 83.18%

## Q3: Type HIV in the PDB website search box on the home page and determine how many HIV-1 protease structures are in the current PDB?

4410

## Q4: Water molecules normally have 3 atoms. Why do we see just one atom per water molecule in this structure?

Each circle represents all 3 atoms in the water molecule.

## Q5: There is a critical “conserved” water molecule in the binding site. Can you identify this water molecule? What residue number does this water molecule have

308? It appears between branches and to be interacting with the ligand.

## Q6: Generate and save a figure clearly showing the two distinct chains of HIV-protease along with the ligand. You might also consider showing the catalytic residues ASP 25 in each chain and the critical water (we recommend “Ball & Stick” for these side-chains). Add this figure to your Quarto document.

![1HSG in yellow and blue with ASP25 on Chain A and B and critical water in green highlight.](1HSG(4).png)

Discussion Topic: Can you think of a way in which indinavir, or even larger ligands and substrates, could enter the binding site?

Maybe if the protein changed in conformation to make the space even larger and allow the ligand in?

```{r}
library(bio3d)
pdb <- read.pdb("1hsg")
pdb
```

## Q7: How many amino acid residues are there in this pdb object?

198

## Q8: Name one of the two non-protein residues?

HOH or MK1

## Q9: How many protein chains are in this structure?

2

```{r}
adk <- read.pdb("6s36")
adk
# Perform flexiblity prediction
m <- nma(adk)
plot(m)
# mktrj() section skipped for PDF
```

## Q10. Which of the packages above is found only on BioConductor and not CRAN?

msa

## Q11. Which of the above packages is not found on BioConductor or CRAN?

bio3d-view

## Q12. True or False? Functions from the devtools package can be used to install packages from GitHub and BitBucket?

TRUE

```{r}
library(bio3d)
aa <- get.seq("1ake_A")
```

## Q13. How many amino acids are in this sequence, i.e. how long is this sequence?

214

```{r}
# Blast or hmmer search 
b <- blast.pdb(aa)
# Plot a summary of search results
hits <- plot(b)
# List out some 'top hits'
head(hits$pdb.id)
# Download related PDB files
files <- get.pdb(hits$pdb.id, path="pdbs", split=TRUE, gzip=TRUE)

# Align related PDBs
pdbs <- pdbaln(files, fit = TRUE, exefile="msa")
# Vector containing PDB codes for figure axis
ids <- basename.pdb(pdbs$id)
```

```{r}
# Draw schematic alignment
# plot(pdbs, labels=ids)
# omitting running this due to error "figure margins too large"
```

```{r}
anno <- pdb.annotate(ids)
unique(anno$source)
anno
```

```{r}
# Perform PCA
pc.xray <- pca(pdbs)
plot(pc.xray)

# Calculate RMSD
rd <- rmsd(pdbs)
# Structure-based clustering
hc.rd <- hclust(dist(rd))
grps.rd <- cutree(hc.rd, k=3)
plot(pc.xray, 1:2, col="grey50", bg=grps.rd, pch=21, cex=1)
# mktrj omitted for pdf
```

```{r}
#Plotting results with ggplot2
library(ggplot2)
library(ggrepel)

df <- data.frame(PC1=pc.xray$z[,1], 
                 PC2=pc.xray$z[,2], 
                 col=as.factor(grps.rd),
                 ids=ids)

p <- ggplot(df) + 
  aes(PC1, PC2, col=col, label=ids) +
  geom_point(size=2) +
  geom_text_repel(max.overlaps = 20) +
  theme(legend.position = "none")
p
```

```{r}
# NMA of all structures
modes <- nma(pdbs)
plot(modes, pdbs, col=grps.rd)
```

## Q14. What do you note about this plot? Are the black and colored lines similar or different? Where do you think they differ most and why?

The black lines are definitely different from the colored lines, especially around the 50 and 120-160 residues. This might because of the difference in conformations producing the 2 different (B&W vs. colored) plots. The residues mentioned indicate nucleotide-binding site regions that change the most in displacement during nucleotide binding.
